extensions/amp-sidebar/1.0/component.js

```diff
@@ -186,49 +187,74 @@ export {Sidebar};
  */
 export function SidebarToolbar({
   toolbar: mediaQueryProp,
-  toolbarTarget,
+  toolbarTarget: toolbarTargetProp,
   children,
   ...rest
 }) {
-  const ref = useRef();
-  const [matches, setMatches] = useState(false);
+  const ref = useRef(null);
+  const [mediaQuery, setMediaQuery] = useState(null);
+  const [toolbarTarget, setToolbarTarget] = useState(null);
   const [targetEl, setTargetEl] = useState(null);
 
   useEffect(() => {
-    const window = ref.current?.ownerDocument?.defaultView;
-    if (!window) {
+    const doc = ref.current?.ownerDocument;
+    if (!doc) {
       return;
     }
 
-    const mediaQueryList = window.matchMedia(mediaQueryProp);
-    const updateMatches = () => setMatches(mediaQueryList.matches);
-    mediaQueryList.addEventListener('change', updateMatches);
-    setMatches(mediaQueryList.matches);
-    return () => mediaQueryList.removeEventListener('change', updateMatches);
+    const sanitizedToolbarTarget = escapeCssSelectorIdent(toolbarTargetProp);
+    setToolbarTarget(sanitizedToolbarTarget);
+    setTargetEl(doc.getElementById(sanitizedToolbarTarget));
+  }, [toolbarTargetProp]);
+
+  useEffect(() => {
+    const win = ref.current?.ownerDocument?.defaultView;
+    if (!win) {
+      return;
+    }
+    setMediaQuery(sanitizeMediaQuery(win, mediaQueryProp));
   }, [mediaQueryProp]);
 
   useEffect(() => {
-    const document = ref.current?.ownerDocument;
-    if (!document) {
+    const element = ref.current;
+    const doc = ref.current?.ownerDocument;
+    if (!doc || !targetEl || mediaQuery == null) {
       return;
     }
 
-    const selector = `#${CSS.escape(toolbarTarget)}`;
-    const newTargetEl = document.querySelector(selector);
-    setTargetEl(newTargetEl);
-  }, [toolbarTarget]);
+    const clone = element.cloneNode(true);
+    const style = doc.createElement('style');
+    style./*OK*/ textContent =
+      `#${toolbarTarget}{display: none;}` +
+      `@media ${mediaQuery}{#${toolbarTarget}{display: initial;}}`;
+
+    targetEl.appendChild(clone);
+    targetEl.appendChild(style);
+    return () => {
+      if (targetEl) {
+        targetEl.removeChild(clone);
+        targetEl.removeChild(style);
+      }
+    };
+  }, [mediaQuery, toolbarTarget, targetEl]);
 
   return (
-    <>
-      <nav
-        ref={ref}
-        toolbar={mediaQueryProp}
-        toolbar-target={toolbarTarget}
-        {...rest}
-      >
-        {children}
-      </nav>
-      {matches && targetEl && createPortal(children, targetEl)}
-    </>
+    <nav
+      ref={ref}
+      toolbar={mediaQueryProp}
+      toolbar-target={toolbarTargetProp}
+      {...rest}
+    >
+      {children}
+    </nav>
   );
 }
+
+/**
+ * @param {!Window} win
+ * @param {string|undefined} query
+ * @return {boolean}
```

Comment:
```suggestion
 * @return {string}
```

Reply:
Thanks, just caught this myself and updated!